liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/AlertDialogDTO.kt

```diff
@@ -0,0 +1,382 @@
+                },
+                modifier = modifier,
+                dismissButton = dismissButton?.let {
+                    {
+                        PhxLiveView(dismissButton, pushEvent, composableNode, null)
+                    }
+                },
+                icon = icon?.let {
+                    {
+                        PhxLiveView(icon, pushEvent, composableNode, null)
+                    }
+                },
+                title = title?.let {
+                    {
+                        PhxLiveView(title, pushEvent, composableNode, null)
+                    }
+                },
+                text = content?.let {
+                    {
+                        PhxLiveView(content, pushEvent, composableNode, null)
+                    }
+                },
+                shape = shape ?: AlertDialogDefaults.shape,
+                containerColor = containerColor ?: AlertDialogDefaults.containerColor,
+                iconContentColor = iconContentColor ?: AlertDialogDefaults.iconContentColor,
+                titleContentColor = titleContentColor ?: AlertDialogDefaults.titleContentColor,
+                textContentColor = textContentColor ?: AlertDialogDefaults.textContentColor,
+                tonalElevation = tonalElevation ?: AlertDialogDefaults.TonalElevation,
+                properties = dialogProperties,
+            )
+        } else {
+            AlertDialog(
+                onDismissRequest = dismissEvent,
+                modifier = modifier,
+                properties = dialogProperties,
+            ) {
+                composableNode?.children?.forEach {
+                    PhxLiveView(it, pushEvent, composableNode, null, this)
+                }
+            }
+        }
+    }
+
+    internal class Builder : ComposableBuilder() {
+        var dismissEvent: () -> Unit = {}
+            private set
+        var dialogProperties: DialogProperties = DialogProperties()
+            private set
+        var shape: Shape? = null
+            private set
+        var containerColor: Color? = null
+            private set
+        var iconContentColor: Color? = null
+            private set
+        var titleContentColor: Color? = null
+            private set
+        var textContentColor: Color? = null
+            private set
+        var tonalElevation: Dp? = null
+            private set
+
+        private var dismissOnBackPress: Boolean = true
+        private var dismissOnClickOutside: Boolean = true
+        private var securePolicy: SecureFlagPolicy = SecureFlagPolicy.Inherit
+        private var usePlatformDefaultWidth: Boolean = true
+        private var decorFitsSystemWindows: Boolean = true
+
+        /**
+         * Whether the dialog can be dismissed by pressing the back button. If true, pressing the
+         * back button will call `dismissEvent` event. Default value is true.
+         * ```
+         * <AlertDialog dismissOnBackPress="true">...</AlertDialog>
```

Comment:
In the SwiftUI client, we were transforming attributes to kebab-case to match attributes like `phx-click` and `phx-value`. So this would become `dismiss-on-back-press`.

@bcardarella Should we change one client to behave a certain way?

Reply:
kebab-case is the correct way

Reply:
I suggest opening an issue for that, because I will have to change all the existing components (not related with this PR).
Could please do that @carson-katri ?

Reply:
https://github.com/liveview-native/liveview-client-jetpack/issues/169